### PR TITLE
ci: allow running MacOS builds on M1 Macs

### PR DESCRIPTION
**Pull request by jakubgs**
This depends on installing Qt via Brew, but that creates a version mismatch, since it's 5.15.8 and not 5.15.2, which is not optimal but works for now.

In the long term we should probably look into using Nix, or maybe aqt will support M1 Macs, but this is not great.

Depends on:
- https://github.com/status-im/infra-ci/commit/54408b41 - macos-brew: symlink /opt/homebrew on x86_64
- https://github.com/status-im/infra-ci/commit/39d4fdef - ci-slave-macos: install qt@5 for desktop M1 builds

Resolves: https://github.com/status-im/status-desktop/issues/9984

### CODE REVIEW

**Comment by jakubgs** [comment]
Looks like it actually worked:
https://ci.status.im/job/status-desktop/job/branches/job/macos/job/aarch64/job/PR-10210/2/

I don't like installing Qt via Brew, since it creates a version mismatch, since it's `5.15.8` and not `5.15.2`, but it seems to work.
In the long term we should probably look into using Nix, or maybe `aqt` will support M1 Macs, but this is not great.

**Review by alexjba — APPROVED** [review]
LGTM! 👍

**Comment by anastasiyaig** [comment]
@jakubgs i added `do not merge` label , please remove it when u are done

File: ci/Jenkinsfile.windows [hunk 1]
```diff
@@ -100,3 +100,10 @@ pipeline {
     cleanup { sh './scripts/clean-git.sh' }
   }
 }
+
+def getArch() {
+    def tokens = Thread.currentThread().getName().split('/')
+    for (def arch in ['x86_64', 'aarch64']) {
+      if (tokens.contains(arch)) { return arch }
+    }
+}
```
Comment (caybro) [inline ci/Jenkinsfile.windows:109]: This looks like it could be extracted to a shared place

Comment (jakubgs) [reply ci/Jenkinsfile.windows:109]: No, it can't. Because you can't use functions from libraries in the `environment` or `agent` contexts.

Comment (jakubgs) [reply ci/Jenkinsfile.windows:109]: Now, by library I mean what we load with the `library` directive, but in theor we could have an explicit:
```groovy
def common = load('common.groovy')
```
And then use that. It should work.

Comment (jakubgs) [reply ci/Jenkinsfile.windows]: No, I don't think I can use `load` outside of a `steps` and `script` contexts:
```
org.jenkinsci.plugins.workflow.steps.MissingContextVariableException: Required context class hudson.FilePath is missing
Perhaps you forgot to surround the code with a step that provides this, such as: node, dockerNode
```

**Review by stefandunca — APPROVED** [review]
I tried the #10 build, which works fine at first glance. I'm really looking forward for the first release with arm for mac.

image

**Pull request merged**